modules/git/commit.go

```diff
@@ -306,6 +306,24 @@ func (c *Commit) HasFile(filename string) (bool, error) {
 	return true, nil
 }
 
+// GetFileContent reads a file content as a string or returns false if this was not possible
+func (c *Commit) GetFileContent(filename string) (string, bool) {
+	entry, err := c.GetTreeEntryByPath(filename)
+	if err != nil {
+		return "", false
+	}
+	r, err := entry.Blob().DataAsync()
```

Comment:
It would be a good idea to limit the size of a file that might get read completely.

Reply:
The file should be very big.

Reply:
Agree with @zeripath .  We have `util.ReadAtMost` to read size limited bytes.

Reply:
... just because in normal use it is not big does not mean that a malicious user won't make it huge.

Reply:
We also need to consider symlinks here.

Reply:
> We also need to consider symlinks here.

Good question ....  

For me, supporting symlinks or not, either is fine to me. 

IMO it's reasonable to only support normal files to keep code simple, since the symlinks in git is not a popular feature among git users. We could just document this behavior: do not use symlinks for templates 😊